src/form/widget/providers/hiddenBox.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 Open Assessment Technologies SA ;
+ */
+/**
+ * Defines a hiddenBox widget
+ * @author Jean-Sébastien Conan <jean-sebastien@taotesting.com>
+ */
+
+import _ from 'lodash';
+import __ from 'i18n';
+import hiddenBoxTpl from 'ui/form/widget/tpl/hiddenBox';
+
+/**
+ * @typedef {widgetConfig} hiddenBoxConfig Defines the config entries available to setup a hiddenBox form widget
+ * @property {Object} [confirmation] - confirmation label, uri and value
+ */
+
+/**
+ * Defines the provider for a hiddenBox widget.
+ *
+ * @example
+ * import widgetHiddenBoxProvider from 'ui/form/widget/providers/hiddenBox';
+ * widgetFactory.registerProvider('hiddenBox', widgetHiddenBoxProvider);
+ * const widget = widgetFactory(container, {
+ *     widget: 'hiddenBox'
+ *     uri: 'hb',
+ *     label: 'Password',
+ *     confirmation: {
+ *         label: 'Please confirm'
+ *     }
+ * });
+ */
+const widgetHiddenBoxProvider = {
+    /**
+     * Initialize the widget.
+     * @param {hiddenBoxConfig} config
+     */
+    init(config) {
+        // the type will be reflected to the HTML markup
+        config.widgetType = 'hidden-box';
+
+        // config for the confirmation field
+        config.confirmation = _.defaults(config.confirmation || {}, {
```

Comment:
Maybe the confirmation needs to be optional.
It can be also for a future improvement

Reply:
Do you mean the widget could be either single password input or confirmed password input?
Should not it be better then to have another component that only supplies a single password?